code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -91,20 +92,37 @@
 	door_obj.icon_state = "[icon_door || icon_state]_door"
 	is_animating_door = TRUE
 	var/num_steps = door_anim_time / world.tick_lag
-	for(var/I in 0 to num_steps)
-		var/angle = door_anim_angle * (closing ? 1 - (I/num_steps) : (I/num_steps))
-		var/matrix/M = get_door_transform(angle)
-		var/door_state = angle >= 90 ? "[icon_door_override ? icon_door : icon_state]_back" : "[icon_door || icon_state]_door"
-		var/door_layer = angle >= 90 ? FLOAT_LAYER : ABOVE_MOB_LAYER
-
-		if(I == 0)
-			door_obj.transform = M
-			door_obj.icon_state = door_state
-			door_obj.layer = door_layer
-		else if(I == 1)
-			animate(door_obj, transform = M, icon_state = door_state, layer = door_layer, time = world.tick_lag, flags = ANIMATION_END_NOW)
-		else
-			animate(transform = M, icon_state = door_state, layer = door_layer, time = world.tick_lag)
+	if(is_crate)
```

Comment:
Just override the proc on the crate subtype.

Reply:
yeah thats definitively better i tried that at the beginning but i think i messed something up initially and went with the if(crate) and just never changed it back